HEATDamage.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Text;
+using System.Diagnostics;
+
+namespace PhysicalDamage.Core
+{
+    class HEATDamage:Damage
+    {
+        public struct HEATData
+        {
+            public float Pierce;
+
+            public float ArmorHEATFactor;
+
+            public float HealthHEATFactor;
+        }
+
+        private HEATData data;
+
+        public HEATDamage(HEATData d,Damage.Target t):base(EDamageTypes.HEAT, t)
+        {
+            data = d;
+        }
+
+        override public float DealDamage()
+        {
+            // No air friction for HEAT round as it detonate on surface of armor
+
+            // Calculate effects of ERA
+            UpdateTargetERA(data.Pierce);
+            data.Pierce = CalculatePostERAPierce(data.Pierce,target.ERAData.HEATFractionMultiplier);
+
+            // Armor degradation
+            UpdateTargetArmor(data.Pierce);
+
+            // Calculate final damage
+            float damage2HP = Math.Max(0.0f, data.Pierce - target.Armor)*data.HealthHEATFactor;
+            return damage2HP;
+        }
+
+        private static float CalculatePostERAPierce(float pierce, float eraFractionMultiplier)
+        {
+            float finalPierce = pierce*(1-eraFractionMultiplier);
+            return finalPierce;
+        }
+        
+        #region UpdateValues
+
+        private void UpdateTargetERA(float piercePostAttenuation)
```

Comment:
Instead of using delegate functions see other comments I've made. In my opinion, this will lead to spaghetti code.